File: app/src/main/java/com/example/paiseslistviewjavaunidad2/MainActivity.java
package com.example.paiseslistviewjavaunidad2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private ListView spnPaises;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        spnPaises = findViewById(R.id.spnPaises);
        ArrayAdapter<String> Adaptador = new ArrayAdapter<String> (MainActivity.this, android.R.layout.simple_expandable_list_item_1,getResources().getStringArray(R.array.paises));
        spnPaises.setAdapter(Adaptador);
        spnPaises.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener(){

            //SE HACE LA CONSULTA DEL PAIS
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
                Toast.makeText(MainActivity.this,"Seleccione el pais " + adapterView.getItemAtPosition(i).toString(), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }

        });
    }
}

File: Practica2Kotlin/app/src/main/java/com/example/practica2kotlin/MainActivity.kt
package com.example.practica2kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView


class MainActivity : AppCompatActivity() {

    private lateinit var editTextWeight: EditText
    private lateinit var editTextHeight: EditText
    private lateinit var buttonCalculate: Button
    private lateinit var btnLimpiar: Button
    private lateinit var btnCerrar: Button
    private lateinit var textViewResult: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnLimpiar = findViewById(R.id.btnLimpiar)
        btnCerrar = findViewById(R.id.btnCerrar)
        editTextWeight = findViewById(R.id.editTextWeight)
        editTextHeight = findViewById(R.id.editTextHeight)
        buttonCalculate = findViewById(R.id.buttonCalculate)
        textViewResult = findViewById(R.id.textViewResult)

        buttonCalculate.setOnClickListener {
            calculateBMI()
        }

        btnLimpiar.setOnClickListener {
            Limpiar()
        }

        btnCerrar.setOnClickListener {
            finish()
        }
    }

    private fun calculateBMI() {
        val weightStr = editTextWeight.text.toString()
        val heightStr = editTextHeight.text.toString()

        if (weightStr.isEmpty() || heightStr.isEmpty()) {
            textViewResult.text = "Por favor inserte Peso y Altura"
            return
        }

        val weight = weightStr.toFloat()
        val height = heightStr.toFloat()

        val bmi = weight / (height * height)

        val result = when {
            bmi < 18.5 -> "Bajo de Peso"
            bmi < 25 -> "Peso Normal"
            bmi < 30 -> "Sobrepeso"
            else -> "Obesidad"
        }

        val bmiResult = "IMC: ${"%.2f".format(bmi)}\n$result"
        textViewResult.text = bmiResult
    }

    private fun Limpiar() {
        editTextWeight.text.clear()
        editTextHeight.text.clear()
        textViewResult.text = ""
    }
}